src/ecdsa.rs

```diff
@@ -66,17 +72,40 @@ impl Signature {
     }
 
     /// Verify an ECDSA signature
-    pub fn verify(self, ctx: &Context, hash: &[u8], pub_key: &PubKey) -> bool {
+    pub fn verify(&self, ctx: &Context, hash: &[u8], pub_key: &PubKey) -> bool {
         1 == unsafe {
             secp256k1_ecdsa_verify(ctx.context, &self.signature, hash.as_ptr(), &pub_key.key)
         }
     }
+
+    /// Return the underlying bytes within a signature
+    pub fn to_bytes(&self) -> [u8; 64] {
+        self.signature.data
+    }
+}
+
+impl From<[u8; 64]> for Signature {
+    /// Create a signature given an array of signed data.
+    fn from(data: [u8; 64]) -> Self {
+        Self {
+            signature: secp256k1_ecdsa_signature { data },
```

Comment:
Here's the func to parse from the compact 64 byte repr:

https://github.com/Trust-Machines/secp256k1/blob/6d7b258680503f929f87e2936febb6d5777a0679/include/secp256k1.h#L486